### PR TITLE
Indiana Earned Income Credit

### LINKED ISSUE #2478: Indiana Earned Income Credit
Reference: https://forms.in.gov/Download.aspx?id=15174

### PR DESCRIPTION
**Pull request by SirMalamute**
Fixes #2478

### CODE REVIEW

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = "Child-level eligiblity for parents filing Indiana EIC"
+    documentation = "Whether a child whose parent filing for Indiana EIC meets the demographic requirements."
+    defined_for = StateCode.ID
+
+    def formula(person, period, parameters):
+        is_19 = person("age", period) < 19
+        is_24 = person("age", period) < 24
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py:14]: 19 & 24 need to be parameters

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from policyengine_us.model_api import *
+
+
+class is_child_eligible_eitc(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py:4]:
```suggestion
class in_eitc_eligible_child(Variable):
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = "Child-level eligiblity for parents filing Indiana EIC"
+    documentation = "Whether a child whose parent filing for Indiana EIC meets the demographic requirements."
+    defined_for = StateCode.ID
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py:10]:
```suggestion
    defined_for = StateCode.IN
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eitc.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from policyengine_us.model_api import *
+
+
+class in_eitc(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eitc.py:4]:
```suggestion
class in_eic(Variable):
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml:1]: change folder name to eic -> all folders

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+description: Indiana EIC is 10% of Federal EIC
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml:1]:
```suggestion
description: Indiana matches this percentage of the federal EITC.
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+values:
+    2022-01-01: 0.1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml:3]: need 2021 value

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+metadata:
+  unit: /1
+  period: year
+  label: Indiana unified elderly tax credit married one aged
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml:7]:
```suggestion
  label: Indiana eic match
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+description: Indiana EIC is 10% of Federal EIC
+values:
+    2022-01-01: 0.1
+metadata:
+  unit: /1
+  period: year
+  label: Indiana unified elderly tax credit married one aged 
+  reference:
+    - title: Indiana Earned Income Credit 2022
+      href: https://forms.in.gov/Download.aspx?id=15432
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/match.yaml:10]: need 2021 reference

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+description: Indiana has income eligibility for EIC, based on the number of children. AGI (Indiana modified, but usually federal) needs to be less than these values depending on num. of children.
+metadata:
+  type: single_amount
+  threshold_unit: child
+  amount_unit: currency-USD
+  label: Indiana unified elderly tax credit married one aged 
+  reference:
+    - title: Indiana Earned Income Credit 2022
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml:8]: need 2021 reference

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml [hunk 10]
```diff
@@ -0,0 +1,25 @@
+description: Indiana has income eligibility for EIC, based on the number of children. AGI (Indiana modified, but usually federal) needs to be less than these values depending on num. of children.
+metadata:
+  type: single_amount
+  threshold_unit: child
+  amount_unit: currency-USD
+  label: Indiana unified elderly tax credit married one aged
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml:6]:
```suggestion
  label: Indiana eic income eligibility
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_is_child_eligible_eitc.py:1]: need test case here as well

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml [hunk 11]
```diff
@@ -0,0 +1,25 @@
+description: Indiana has income eligibility for EIC, based on the number of children. AGI (Indiana modified, but usually federal) needs to be less than these values depending on num. of children.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/child_income_bracket_eligibility.yaml:1]:
```suggestion
description: Indiana provides the EIC for filers with income below this threshold, based on number of children.
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml:1]: need to add chnagelog

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+description: Age threshold to be eligible for as a qualifying child in IN EIC without being a full time student
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml:1]:
```suggestion
description: Indiana provides the EIC for filers with children below this age threshold.
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml:1]: same in other age yaml

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+values:
+  2021-01-01: 19
+metadata:
+  reference:
+    - title: Qualifying Child Rules | IRS
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml:6]: add a reference from indiana

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml:6]: both yaml

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+      href: https://www.irs.gov/credits-deductions/individuals/earned-income-tax-credit/qualifying-child-rules
+  unit: year
+  label: IN EIC Qualifying Child
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml:9]:
```suggestion
  label: Indiana EIC non-student child age threshold 
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age_child.yaml:1]: make a folder called `age` with 2 files in it

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml [hunk 15]
```diff
@@ -0,0 +1,27 @@
+description: Indiana provides the EIC for filers with income below this threshold, based on number of children.
+metadata:
+  type: single_amount
+  threshold_unit: child
+  amount_unit: currency-USD
+  label: Indiana eic income eligibility 
+  reference:
+    - title: Indiana Earned Income Credit 2022
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml:8]: add a legal reference

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml:8]: all yaml files

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic_eligible_child.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic_eligible_child.py:1]: need to create a unit test for this variable as well

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_child.yaml [hunk 16]
```diff
@@ -0,0 +1,13 @@
+description: Age threshold to be eligible for as a qualifying child in IN EIC without being a full time student
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_child.yaml:1]:
```suggestion
description: Indiana provides the EITC for filers with children below this age threshold.
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_full_time_student.yaml [hunk 17]
```diff
@@ -0,0 +1,14 @@
+description: Age threshold to be eligible for as a qualifying child in IN EIC as a full time student
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_full_time_student.yaml:1]:
```suggestion
description: Indiana provides the EITC for filers with children who are full-time students below this age threshold.
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_child.yaml [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  reference:
+    - title: Qualifying Child Rules | IRS
+      href: https://www.irs.gov/credits-deductions/individuals/earned-income-tax-credit/qualifying-child-rules
+    - title: Indiana Earned Income Credit 2022
+      href: https://forms.in.gov/Download.aspx?id=15432 
+    - title: Indiana Earned Income Credit 2021
+      href: https://forms.in.gov/Download.aspx?id=15174
+  unit: year
+  label: Indiana provides the EIC for filers with children who are not full time students below this age threshold.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_child.yaml:13]:
```suggestion
  label: Indiana EIC child age threshold
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_full_time_student.yaml [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+  reference:
+    - title: Qualifying Child Rules | IRS
+      href: https://www.irs.gov/credits-deductions/individuals/earned-income-tax-credit/qualifying-child-rules
+    - title: Indiana Earned Income Credit 2022
+      href: https://forms.in.gov/Download.aspx?id=15432 
+    - title: Indiana Earned Income Credit 2021
+      href: https://forms.in.gov/Download.aspx?id=15174
+
+  unit: year
+  label: Indiana provides the EIC for filers with children who are full time students below this age threshold.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_full_time_student.yaml:14]:
```suggestion
  label: Indiana EITC full-time student age threshold
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_child.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_child.yaml:1]: rename to child.yaml

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_full_time_student.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/age/age_full_time_student.yaml:1]: rename to full_time_student.yaml

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml [hunk 20]
```diff
@@ -0,0 +1,27 @@
+description: Indiana provides the EIC for filers with income below this threshold, based on number of children.
+metadata:
+  type: single_amount
+  threshold_unit: child
+  amount_unit: currency-USD
+  label: Indiana eic income eligibility
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml:6]:
```suggestion
  label: Indiana EITC income eligibility 
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml [hunk 21]
```diff
@@ -0,0 +1,27 @@
+description: Indiana provides the EIC for filers with income below this threshold, based on number of children.
+metadata:
+  type: single_amount
+  threshold_unit: child
+  amount_unit: currency-USD
+  label: Indiana eic income eligibility 
+  reference:
+    - title: Indiana Earned Income Credit 2022
+      href: https://forms.in.gov/Download.aspx?id=15432 
+    - title: Indiana Earned Income Credit 2021
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml:10]: need page for the tax form where we can find the value

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/match.yaml [hunk 22]
```diff
@@ -0,0 +1,13 @@
+description: Indiana matches this percentage of the federal EITC.
+values:
+    2021-01-01: 0.1
+metadata:
+  unit: /1
+  period: year
+  label: Indiana eic match
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eic/match.yaml:7]:
```suggestion
  label: Indiana EITC match
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml (hunk 21 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/in/tax/income/credits/eic/income_threshold.yaml:10]: all yaml files

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic_eligible_child.py [hunk 23]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class in_eic_eligible_child(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic_eligible_child.py:4]:
```suggestion
class in_eitc_eligible_child(Variable):
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic_eligible_child.py [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = "Child-level eligiblity for parents filing Indiana EIC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic_eligible_child.py:8]:
```suggestion
    label = "Child-level eligiblity for parents filing Indiana EITC"
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic.py [hunk 25]
```diff
@@ -0,0 +1,26 @@
+from policyengine_us.model_api import *
+
+
+class in_eic(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic.py:4]:
```suggestion
class in_eitc(Variable):
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic.py:1]: need to switch all names to EITC - as it is a branch of the earned income tax credit - for uniformity across the model

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic.py [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+    value_type = float
+    entity = TaxUnit
+    definition_period = YEAR
+    label = "Indiana EIC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eic.py:8]:
```suggestion
    label = "Indiana EITC"
```

File: policyengine_us/tests/policy/baseline/gov/states/in/tax/income/credits/eitc/in_eitc_eligible_child.yaml [hunk 27]
```diff
@@ -0,0 +1,44 @@
+- name: Over 18 and not disabled and student
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/in/tax/income/credits/eitc/in_eitc_eligible_child.yaml:1]: need to add state_code: IN to all tests

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: after all changes are done, need to run `make format`

File: policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eitc.py [hunk 28]
```diff
@@ -0,0 +1,26 @@
...
+    value_type = float
+    entity = TaxUnit
+    definition_period = YEAR
+    label = "Indiana EIC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/credits/eitc/in_eitc.py:8]:
```suggestion
    label = "Indiana EITC"
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/age/child.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/eitc/age/child.yaml:1]: The form looks like it uses the federal EITC definition. Please find the Indiana legal code to confirm, and if so, we can delete these parameters and point to `is_eitc_qualifying_child`

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Per the [legal code](https://iga.in.gov/laws/2022/ic/titles/6#6-3.1-21), we can represent this program with only the 10% match parameter. Please remove other parameters and examine the state legal code before developing in the future to avoid unnecessary work.

**Review by PavelMakarchuk — APPROVED** [review]

**Comment by MaxGhenis** [comment]
Please resolve all outstanding comment threads and tag me for a review

**Comment by PavelMakarchuk** [comment]
> Per the [legal code](https://iga.in.gov/laws/2022/ic/titles/6#6-3.1-21), we can represent this program with only the 10% match parameter. Please remove other parameters and examine the state legal code before developing in the future to avoid unnecessary work.

resolved

**Comment by PavelMakarchuk** [comment]
> Please resolve all outstanding comment threads and tag me for a review

all has been resolved

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
I may have spoke too soon previously. This looks like it matches an earlier iteration of the federal EITC. Please move `in_eitc.yaml` under a new folder `eitc` (matching the variable folder structure) and add some integration tests to `eitc/integration.yaml` going through a sample filer's IN EITC calculation based on the tax form.

If we need to use an old version of the federal EITC, we'll want to place this PR on hold until we can add historical EITC rules (which it would be great if you could do, in another PR). From there we have tools for accessing previous years' logic.

**Comment by MaxGhenis** [comment]
Hi @SirMalamute - since this ended up being a trickier one I'm going to suggest we close it in favor of @martinholmer's #2815. Thanks for your work here!

**Pull request closed without merging**